### ISSUE TITLE
network::cisco::standard::snmp::plugin set critical and warning thresholds

### ISSUE THREAD

**Issue by dgoudet** [centreon/centreon-plugins#980]
Hi,

I am using "network::cisco::standard::snmp::plugin" plugin to monitor cisco ASR1001X and C2960X devices.
On some devices fiber quality monitoring status is WARNING or CRITICAL.
In nominal case WARNING and CRITICAL threshold are fetched from dedicated threshold OIDs defined in Cisco MIB device.
In my environment these thresholds are not convenient, for example one status:
**WARNING: Sensor 'subslot 0/0 transceiver 4 Rx Power Sensor/1147' is -2.8 dBm**

I would like to set my custom CRITICAL and WARNING thresholds values to prevent false positive alert on this check.

What is the better way to set this threshold?

--critical and --warning parameters on script seems to enable threshold overriding but after some tests nothing is working.

On example below by setting CRITICAL and WARNING with threshold out of current value range (20:40, 25:36) we may get status CRITICAL or WARNING  on temperature sensors but nothing occurs.

**sudo -u centreon-engine /usr/lib64/nagios/plugins/centreon-plugins/centreon_plugins.pl --plugin=network::cisco::standard::snmp::plugin --mode=environment --hostname=XXX --snmp-version='3' --snmp-community='public' --snmp-username xxx --privpassphrase xxxx --authpassphrase xxxx --authprotocol SHA --privprotocol AES  --critical='temperature,.\*,20:40' --warning='temperature,.\*,25:36'**

_WARNING: Sensor 'subslot 0/0 transceiver 4 Rx Power Sensor/1147' is -2.8 dBm | 'sensor.amperes_PEM Iout'=5A;;;; 'sensor.voltsDC_PEM Vout'=12V;;;; 'sensor.voltsAC_PEM Vin'=231V;;;; 'sensor.celsius_Temp: PEM In'=27C;0:79;0:99;; 'sensor.celsius_Temp: PEM Out'=25C;0:79;0:99;; 'sensor.celsius_Temp: PEM Int'=32C;0:79;0:99;; 'sensor.amperes_PEM Iout'=5A;;;; 'sensor.voltsDC_PEM Vout'=12V;;;; 'sensor.voltsAC_PEM Vin'=232V;;;; 'sensor.celsius_Temp: PEM In'=27C;0:79;0:99;; 'sensor.celsius_Temp: PEM Out'=23C;0:79;0:99;; 'sensor.celsius_Temp: PEM Int'=29C;0:79;0:99;; 'sensor.celsius_subslot 0/0 transceiver 4 Temperature Sensor'=23.808C;-4.999:84.999;-8.999:88.999;; 'sensor.voltsDC_subslot 0/0 transceiver 4 Supply Voltage Sensor'=3.3182V;3.1001:3.4999;3.0001:3.5999;; 'sensor.amperes_subslot 0/0 transceiver 4 Bias Current Sensor'=0.00852A;0.001001:0.049999;0.000501:0.059999;; 'sensor.dBm_subslot 0/0 transceiver 4 Tx Power Sensor'=-5.6dBm;-9.4:-3.1;-13.4:0.9;; 'sensor.dBm_subslot 0/0 transceiver 4 Rx Power Sensor'=-2.8dBm;-18.9:-3.1;-22.9:0.9;; 'sensor.celsius_subslot 0/1 temperature Sensor 0'=24C;-4:69;0:89;; 'sensor.celsius_subslot 0/1 temperature Sensor 1'=26C;-4:69;0:89;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 0'=3.31V;3.136:3.464;3.07:3.53;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 1'=2.502V;2.376:2.624;2.326:2.674;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 2'=1.502V;1.426:1.574;1.396:1.604;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 3'=1.201V;1.141:1.259;1.117:1.283;; 'sensor.celsius_subslot 0/1 transceiver 0 Temperature Sensor'=25.242C;-4.999:84.999;-9.999:89.999;; 'sensor.voltsDC_subslot 0/1 transceiver 0 Supply Voltage Sensor'=3.3368V;3.1401:3.4699;2.9701:3.6299;; 'sensor.amperes_subslot 0/1 transceiver 0 Bias Current Sensor'=0.003398A;0.002401:0.011899;0.002001:0.012749;; 'sensor.dBm_subslot 0/1 transceiver 0 Tx Power Sensor'=-5.7dBm;-9.4:-3.1;-13.4:-0.1;; 'sensor.dBm_subslot 0/1 transceiver 0 Rx Power Sensor'=-7.9dBm;-16.9:-0.1;-20.9:2.9;; 'sensor.celsius_subslot 0/1 transceiver 1 Temperature Sensor'=25.125C;-4.999:84.999;-9.999:89.999;; 'sensor.voltsDC_subslot 0/1 transceiver 1 Supply Voltage Sensor'=3.3398V;3.1401:3.4699;2.9701:3.6299;; 'sensor.amperes_subslot 0/1 transceiver 1 Bias Current Sensor'=0.003534A;0.002401:0.011899;0.002001:0.012749;; 'sensor.dBm_subslot 0/1 transceiver 1 Tx Power Sensor'=-5.7dBm;-9.4:-3.1;-13.4:-0.1;; 'sensor.dBm_subslot 0/1 transceiver 1 Rx Power Sensor'=-7.2dBm;-16.9:-0.1;-20.9:2.9;; 'sensor.voltsDC_VVM 0: VX1'=1.518V;;;; 'sensor.voltsDC_VVM 0: VX2'=0.758V;;;; 'sensor.voltsDC_VVM 0: VX3'=1.003V;;;; 'sensor.voltsDC_VVM 0: VX4'=1.052V;;;; 'sensor.voltsDC_VVM 0: VX5'=0.898V;;;; 'sensor.voltsDC_VVM 0: VP1'=5.112V;;;; 'sensor.voltsDC_VVM 0: VP3'=1.505V;;;; 'sensor.voltsDC_VVM 0: VP4'=1.791V;;;; 'sensor.voltsDC_VVM 0: VH'=11.935V;;;; 'sensor.voltsDC_VVM 1: VX1'=1.201V;;;; 'sensor.voltsDC_VVM 1: VX2'=0.84V;;;; 'sensor.voltsDC_VVM 1: VX3'=0.857V;;;; 'sensor.voltsDC_VVM 1: VX4'=0.997V;;;; 'sensor.voltsDC_VVM 1: VX5'=0.899V;;;; 'sensor.voltsDC_VVM 1: VP2'=3.351V;;;; 'sensor.voltsDC_VVM 1: VP3'=1.804V;;;; 'sensor.voltsDC_VVM 1: VP4'=0.997V;;;; 'sensor.voltsDC_VVM 1: VH'=11.93V;;;; 'sensor.voltsDC_VVM 2: VX1'=1.113V;;;; 'sensor.voltsDC_VVM 2: VX2'=1.103V;;;; 'sensor.voltsDC_VVM 2: VX3'=1.218V;;;; 'sensor.voltsDC_VVM 2: VX4'=2.492V;;;; 'sensor.voltsDC_VVM 2: VX5'=0.965V;;;; 'sensor.voltsDC_VVM 2: VP2'=1.524V;;;; 'sensor.voltsDC_VVM 2: VP3'=1.514V;;;; 'sensor.voltsDC_VVM 2: VP4'=2.493V;;;; 'sensor.voltsDC_VVM 2: VH'=11.94V;;;; 'sensor.voltsDC_VVM 2: AUX1'=0.751V;;;; 'sensor.voltsDC_VVM 2: AUX2'=0.756V;;;; 'sensor.celsius_Temp: sTCAM'=24C;0:64;0:99;; 'sensor.celsius_Temp: Inlet'=19C;0:54;0:79;; 'sensor.celsius_Temp: Outlet'=33C;0:74;0:99;; 'sensor.celsius_Temp: QFP Die'=34C;0:99;0:124;; 'sensor.celsius_Temp: Center'=33C;0:74;0:99;; 'sensor.celsius_Temp: Oct Die'=37C;0:99;0:124;; 'sensor.celsius_Temp: CPU Inlt'=23C;0:74;0:99;; 'sensor.celsius_Temp: CPU VRM'=23C;0:74;0:99;; 'sensor.celsius_Temp: CPU Die'=32C;0:99;0:124;; 'sensor.celsius_Temp: FC FANS'=19C;;;;_

Regards,

**Comment by pkriko** [comment]
Hello,

Can you try this syntax for the temperature : 
--warning="sensor.celsius,.*,25:36" --critical='sensor.celsius,.*,20:40"

**Comment by dgoudet** [comment]
Hello,

Thank you for your help, it works as excepted.

I took temperature/sensors.celcius as example but my initial issue was on sensor.dBm (transceiver) parameter. I used and adapt your threshold configuration with sensor.dBm parameter and it works also:

--warning="sensor.dBm,.,-18.9:-2.5"  --critical='sensor.dBm,.,-3:70'

I need to set threshold on "transceiver Rx" parameter only (not on transceiver Tx), i used "regex" field on --warning parameter but it does not work as expected: threshold is not defined at all in this case.

--warning="sensor,.*Rx.*,-18.9:-2.5" 

Result:
....sensor.dBm_subslot 0/0 transceiver 4 Tx Power Sensor'=-5.6dBm;;-3:70;;...

What is the better way to define the threshold on "transceiver Rx" parameter only?

Hereafter example 

Without set threshold:
**sudo -u centreon-engine /usr/lib64/nagios/plugins/centreon-plugins/centreon_plugins.pl --plugin=network::cisco::standard::snmp::plugin --mode=environment --hostname=XXX --snmp-version='3' --snmp-community='public' --snmp-username xxx --privpassphrase xxxx --authpassphrase xxxx --authprotocol SHA --privprotocol AES**
...
'sensor.dBm_subslot 0/0 transceiver 4 Tx Power Sensor'=-5.6dBm;-9.4:-3.1;-13.4:0.9;; 'sensor.dBm_subslot 0/0 transceiver 4 Rx Power Sensor'=-2.8dBm;-18.9:-3.1;-22.9:0.9;; 'sensor.celsius_subslot 0/1 temperature Sensor 0'=24C;-4:69;0:89;; 'sensor.celsius_subslot 0/1 temperature Sensor 1'=26C;-4:69;0:89;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 0'=3.31V;3.136:3.464;3.07:3.53;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 1'=2.504V;2.376:2.624;2.326:2.674;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 2'=1.501V;1.426:1.574;1.396:1.604;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 3'=1.201V;1.141:1.259;1.117:1.283;; 'sensor.celsius_subslot 0/1 transceiver 0 Temperature Sensor'=25.429C;-4.999:84.999;-9.999:89.999;; 'sensor.voltsDC_subslot 0/1 transceiver 0 Supply Voltage Sensor'=3.3368V;3.1401:3.4699;2.9701:3.6299;; 'sensor.amperes_subslot 0/1 transceiver 0 Bias Current Sensor'=0.003398A;0.002401:0.011899;0.002001:0.012749;; 'sensor.dBm_subslot 0/1 transceiver 0 Tx Power Sensor'=-5.7dBm;-9.4:-3.1;-13.4:-0.1;; 'sensor.dBm_subslot 0/1 transceiver 0 Rx Power Sensor'=-7.9dBm;-16.9:-0.1;-20.9:2.9;; 'sensor.celsius_subslot 0/1 transceiver 1 Temperature Sensor'=25.125C;-4.999:84.999;-9.999:89.999;; 'sensor.voltsDC_subslot 0/1 transceiver 1 Supply Voltage Sensor'=3.3398V;3.1401:3.4699;2.9701:3.6299;; 'sensor.amperes_subslot 0/1 transceiver 1 Bias Current Sensor'=0.003534A;0.002401:0.011899;0.002001:0.012749;; 'sensor.dBm_subslot 0/1 transceiver 1 Tx Power Sensor'=-5.7dBm;-9.4:-3.1;-13.4:-0.1;; 'sensor.dBm_subslot 0/1 transceiver 1 Rx Power Sensor'=-7.2dBm;-16.9:-0.1;-20.9:2.9;;

When i set threshold on all parameters sensor.dBm (it works):

**sudo -u centreon-engine /usr/lib64/nagios/plugins/centreon-plugins/centreon_plugins.pl --plugin=network::cisco::standard::snmp::plugin --mode=environment --hostname=XXX --snmp-version='3' --snmp-community='public' --snmp-username xxx --privpassphrase xxxx --authpassphrase xxxx --authprotocol SHA --privprotocol AES --warning="sensor.dBm,.,-18.9:-2.5"  --critical='sensor.dBm,.,-3:70**
'sensor.dBm_subslot 0/0 transceiver 4 Tx Power Sensor'=-5.6dBm;-18.9:-2.5;-3:70;; 'sensor.dBm_subslot 0/0 transceiver 4 Rx Power Sensor'=-2.8dBm;-18.9:-2.5;-3:70;; 'sensor.celsius_subslot 0/1 temperature Sensor 0'=24C;-4:69;0:89;; 'sensor.celsius_subslot 0/1 temperature Sensor 1'=26C;-4:69;0:89;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 0'=3.31V;3.136:3.464;3.07:3.53;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 1'=2.504V;2.376:2.624;2.326:2.674;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 2'=1.501V;1.426:1.574;1.396:1.604;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 3'=1.201V;1.141:1.259;1.117:1.283;; 'sensor.celsius_subslot 0/1 transceiver 0 Temperature Sensor'=25.429C;-4.999:84.999;-9.999:89.999;; 'sensor.voltsDC_subslot 0/1 transceiver 0 Supply Voltage Sensor'=3.3368V;3.1401:3.4699;2.9701:3.6299;; 'sensor.amperes_subslot 0/1 transceiver 0 Bias Current Sensor'=0.003398A;0.002401:0.011899;0.002001:0.012749;; 'sensor.dBm_subslot 0/1 transceiver 0 Tx Power Sensor'=-5.7dBm;-18.9:-2.5;-3:70;; 'sensor.dBm_subslot 0/1 transceiver 0 Rx Power Sensor'=-7.9dBm;-18.9:-2.5;-3:70;; 'sensor.celsius_subslot 0/1 transceiver 1 Temperature Sensor'=25.125C;-4.999:84.999;-9.999:89.999;; 'sensor.voltsDC_subslot 0/1 transceiver 1 Supply Voltage Sensor'=3.3398V;3.1401:3.4699;2.9701:3.6299;; 'sensor.amperes_subslot 0/1 transceiver 1 Bias Current Sensor'=0.003534A;0.002401:0.011899;0.002001:0.012749;; 'sensor.dBm_subslot 0/1 transceiver 1 Tx Power Sensor'=-5.7dBm;-18.9:-2.5;-3:70;; 'sensor.dBm_subslot 0/1 transceiver 1 Rx Power Sensor'=-7.2dBm;-18.9:-2.5;-3:70;; 

When i try to set threshold on Rx parameters sensor.dBm only (it does not works):

**sudo -u centreon-engine /usr/lib64/nagios/plugins/centreon-plugins/centreon_plugins.pl --plugin=network::cisco::standard::snmp::plugin --mode=environment --hostname=XXX --snmp-version='3' --snmp-community='public' --snmp-username xxx --privpassphrase xxxx --authpassphrase xxxx --authprotocol SHA --privprotocol AES --warning="sensor.dBm,.*Rx.*,-18.9:-2.5"  --critical='sensor.dBm,.,-3:70'**
...
'sensor.dBm_subslot 0/0 transceiver 4 Tx Power Sensor'=-5.6dBm;;-3:70;; 'sensor.dBm_subslot 0/0 transceiver 4 Rx Power Sensor'=-2.8dBm;;-3:70;; 'sensor.celsius_subslot 0/1 temperature Sensor 0'=24C;-4:69;0:89;; 'sensor.celsius_subslot 0/1 temperature Sensor 1'=26C;-4:69;0:89;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 0'=3.31V;3.136:3.464;3.07:3.53;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 1'=2.504V;2.376:2.624;2.326:2.674;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 2'=1.501V;1.426:1.574;1.396:1.604;; 'sensor.voltsDC_subslot 0/1 voltage Sensor 3'=1.201V;1.141:1.259;1.117:1.283;; 'sensor.celsius_subslot 0/1 transceiver 0 Temperature Sensor'=25.429C;-4.999:84.999;-9.999:89.999;; 'sensor.voltsDC_subslot 0/1 transceiver 0 Supply Voltage Sensor'=3.3368V;3.1401:3.4699;2.9701:3.6299;; 'sensor.amperes_subslot 0/1 transceiver 0 Bias Current Sensor'=0.003398A;0.002401:0.011899;0.002001:0.012749;; 'sensor.dBm_subslot 0/1 transceiver 0 Tx Power Sensor'=-5.7dBm;;-3:70;; 'sensor.dBm_subslot 0/1 transceiver 0 Rx Power Sensor'=-7.9dBm;;-3:70;; 'sensor.celsius_subslot 0/1 transceiver 1 Temperature Sensor'=25.125C;-4.999:84.999;-9.999:89.999;; 'sensor.voltsDC_subslot 0/1 transceiver 1 Supply Voltage Sensor'=3.3398V;3.1401:3.4699;2.9701:3.6299;; 'sensor.amperes_subslot 0/1 transceiver 1 Bias Current Sensor'=0.003534A;0.002401:0.011899;0.002001:0.012749;; 'sensor.dBm_subslot 0/1 transceiver 1 Tx Power Sensor'=-5.7dBm;;-3:70;; 'sensor.dBm_subslot 0/1 transceiver 1 Rx Power Sensor'=-7.2dBm;;-3:70;; 
...
